[Task]: Add parameter types getGetter/getSetterCode

mcop1 opened the pull request:
Add types for all parameters, including $language parameter
Removed setter functions for `CalculatedValue`, since they are not used, change return values for getPhpdocInputType/getParameterTypeDeclaration accordingly

mcop1 committed:
[Task]: Add return type for $language parameter in getGetter/getSetterCode for localized fields

mcop1 committed:
[Task]: Add parameter type for $language parameter in getGetter/getSetterCode for localized fields

mcop1 committed:
[Task]: Add return type for $language parameter

mcop1 committed:
[Task]: Add parameter type for $language parameter

jdreesen commented:
The title is wrong, it should be *parameter type* ;)

mcop1 commented:
> The title is wrong, it should be _parameter type_ ;)

Thanks

blankse commented on models/DataObject/ClassDefinition/Data/CalculatedValue.php:
```diff
@@ -510,7 +510,7 @@ public function getSetterCodeLocalizedfields($class)
         $code .= '* @return $this' . "\n";
         $code .= '*/' . "\n";
-        $code .= 'public function set' . ucfirst($key) . '(' . '$' . $key . ', $language = null): static' . "\n";
+        $code .= 'public function set' . ucfirst($key) . '(' . '$' . $key . ', ?string $language = null): static' . "\n";
```
Should we add also the missing parameter type here?
```suggestion
        $code .= 'public function set' . ucfirst($key) . '(' . $this->getParameterTypeDeclaration() . ' $' . $key . ', ?string $language = null): static' . "\n";
```

But this is missing also in the other getSetterCode methods...

mcop1 replied:
I was thinking about doing another pr, but I guess it is ok to do it in the same pr, so thanks for the suggestion!

mcop1 committed:
Update models/DataObject/ClassDefinition/Data/CalculatedValue.php

Co-authored-by: Sebastian Blank <blank@data-factory.net>

mcop1 committed:
Update models/DataObject/ClassDefinition/Data/CalculatedValue.php

Co-authored-by: Sebastian Blank <blank@data-factory.net>

mcop1 committed:
[Task]: Add parameter types getGetter/getSetterCode

mcop1 committed:
Merge remote-tracking branch 'origin/fix_language_parameter_type' into fix_language_parameter_type

# Conflicts:
#	models/DataObject/ClassDefinition/Data/CalculatedValue.php

blankse commented:
Why do empty setters exist for CalculatedValue? Should we remove them completely?

blankse commented:
The methods getParameterTypeDeclaration()/getPhpdocInputType() must return string "mixed"

mcop1 committed:
[Task]: Add parameter types getGetter/getSetterCode
Removed setter functions, changed return values for getPhpdocInputType/getParameterTypeDeclaration for CalculatedValue.php

mcop1 commented:
> The methods getParameterTypeDeclaration()/getPhpdocInputType() must return string "mixed"

I changed the `getSetterCodeXXX` functions to return an empty string and adapted the testcases. I don´t think there is a point in keeping the setters, what do you think?

blankse commented:
I also think that we do not need them.

mcop1 committed:
[Task]: Add parameter types getGetter/getSetterCode
Added upgrade notes

mcop1 committed:
Merge branch '11.x' into fix_language_parameter_type

mcop1 committed:
Merge branch '11.x' into fix_language_parameter_type
